### ISSUE TITLE
missing_docs lint ignores pub consts

### ISSUE THREAD

**Issue by crumblingstatue** [rust-lang/rust#24249]
```rust
//! A lonely crate
#![warn(missing_docs)]
pub const FOO: i32 = 0;
```

This should warn about missing documentation for FOO, but it doesn't.
